src/common/datasource/src/lister.rs

```diff
@@ -70,11 +69,14 @@ impl Lister {
                     .context(error::ListObjectsSnafu { path: &self.path })
             }
             Source::Filename(filename) => {
-                let obj = self
-                    .object_store
-                    .object(&format!("{}{}", self.path, filename));
-
-                Ok(vec![obj])
+                // make sure this file exists
+                let file_full_path = format!("{}{}", self.path, filename);
+                let _ = self.object_store.stat(&file_full_path).await.context(
+                    error::ListObjectsSnafu {
+                        path: &file_full_path,
+                    },
+                )?;
```

Comment:
Is it possible that a file doesn't exist while listing the data source?

Reply:
> Is it possible that a file doesn't exist while listing the data source?

OpenDAL always raises an error if target file path does not exist.